tubes/lua.lua

```diff
@@ -228,12 +228,15 @@ end
 -- Parsing and running --
 -------------------------
 
-local function safe_print(param)
-	local string_meta = getmetatable("")
-	local sandbox = string_meta.__index
-	string_meta.__index = string -- Leave string sandbox temporarily
-	print(dump(param))
-	string_meta.__index = sandbox -- Restore string sandbox
+local safe_print = function() end
+if pipeworks.lua_tube_print_behavior == "log" then
+	safe_print = function(param)
+		local string_meta = getmetatable("")
+		local sandbox = string_meta.__index
+		string_meta.__index = string -- Leave string sandbox temporarily
+		minetest.log("action", string.format("[pipeworks.tubes.lua] print(%s)", dump(param)))
+		string_meta.__index = sandbox -- Restore string sandbox
+	end
```

Comment:
Should we still keep this configurable run time or do you think having it only load time is better?

Like how mesecons allows run time configuration and if any this would be one of those configs I could imagine someone changing without restarting game.

Reply:
Oh, if mesecons allow runtime configuration then we should too. Though Pipeworks cannot modify the configuration table according to the file yet...

Reply:
Done in 9cce442, though if we want it to be truly adjustable, we must change `default_settings.lua` so that the values can be reloaded.

Reply:
To make it configurable runtime you can just do direct call to `minetest.settings.get` like this:

```suggestion
local function safe_print(param)
	if minetest.settings:get("pipeworks_lua_tube_print_behavior") == "log" then
		local string_meta = getmetatable("")
		local sandbox = string_meta.__index
		string_meta.__index = string -- Leave string sandbox temporarily
		minetest.log("action", string.format("[pipeworks.tubes.lua] print(%s)", dump(param)))
		string_meta.__index = sandbox -- Restore string sandbox
	end
end
```

Then you can just use `/set` builtin chat command to change configuration run time.

Reply:
Done in 0023175